web/src/app/util/PageActions.js

```diff
@@ -8,104 +8,91 @@ const PageActionsContext = React.createContext({
 })
 PageActionsContext.displayName = 'PageActionsContext'
 
-export class PageActionContainer extends React.PureComponent {
-  render() {
-    return (
-      <PageActionsContext.Consumer>
-        {({ actions }) => actions}
-      </PageActionsContext.Consumer>
-    )
-  }
+export const PageActionContainer = () => {
+  const { actions } = useContext(PageActionsContext)
+  return actions
 }
 
-export class PageActionProvider extends React.PureComponent {
-  state = {
-    actions: null,
-  }
-
-  _mountCount = 0
-  _mounted = false
-
-  componentDidMount() {
-    this._mounted = true
-    if (this._pending) {
-      this.setActions(this._pending)
-      this._pending = null
-    }
-  }
+export const PageActionProvider = (props) => {
+  const [actions, setActions] = useState(null)
+  const _mountCount = useRef(0)
+  const _mounted = useRef(false)
+  const _pending = useRef(null)
 
-  componentWillUnmount() {
-    this._mounted = false
-    this._pending = false
-    this.setActions.cancel()
-  }
-
-  _setActions = (actions) => {
-    if (!this._mounted) {
-      this._pending = actions
+  const _setActions = (actions) => {
+    if (!_mounted.current) {
+      _pending.current = actions
       return
     }
 
-    this.setState({ actions })
+    setActions(actions)
   }
 
-  setActions = debounce(this._setActions)
+  const debouncedSetActions = debounce(_setActions)
 
-  updateMounted = (mount) => {
+  const updateMounted = (mount) => {
     if (mount) {
-      this._mountCount++
+      _mountCount.current++
     } else {
-      this._mountCount--
+      _mountCount.current--
     }
 
-    if (this._mountCount > 1 && global.console && console.error) {
+    if (_mountCount.current > 1 && global.console && console.error) {
       console.error(
         'PageActions: Found more than one <PageActions> component mounted within the same provider.',
       )
     }
   }
 
-  render() {
-    return (
-      <PageActionsContext.Provider
-        value={{
-          actions: this.state.actions,
-          setActions: this.setActions,
-          trackMount: this.updateMounted,
-        }}
-      >
-        {this.props.children}
-      </PageActionsContext.Provider>
-    )
-  }
+  useEffect(() => {
+    _mounted.current = true
+    if (_pending.current) {
+      debouncedSetActions(_pending.current)
+      _pending.current = false
+    }
+    return () => {
+      _mounted.current = false
+      _pending.current = false
+      debouncedSetActions.cancel()
+    }
+  })
```

Comment:
Adding an empty dependency array at the end will ensure this `useEffect` only runs once on mount and once on unmount -- otherwise, it's going to call `debouncedSetActions` and `.cancel` on every render.
```suggestion
  }, [])
```